alcove/index/language.py

```
@@ -0,0 +1,288 @@
+        except json.JSONDecodeError:
+            data = {}
+        return LanguageDetection(_normalize_language(data.get("language")), provider=self.provider)
+
+    def _post(self, path: str, payload: dict) -> dict:
+        body = json.dumps(payload).encode("utf-8")
+        request = urllib.request.Request(
+            f"{self.base_url}{path}",
+            data=body,
+            headers={"Content-Type": "application/json"},
+            method="POST",
+        )
```

Ruff (0.15.12)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)